githubapi-frontend/src/app/shared/service/service.service.ts

```diff
@@ -1,22 +1,21 @@
 import { Injectable } from '@angular/core';
 import { HttpClient } from '@angular/common/http';
 import { Observable } from 'rxjs';
-import { tap } from 'rxjs';
 import { environment } from 'src/environments/environment';
+import { User } from '../interface/user.model';
+import { SearchResultUser } from '../interface/search-result-user.model';
 
 @Injectable({
   providedIn: 'root'
 })
 export class ServiceService {
 
   private readonly API = `${environment.API}`;
-  users = []
+  users!: User;
```

Comment:
Avaliar aonde esta propriedade está sendo utilizada e o pq. 
Não existe a necessidade de por o `!` entre a propriedade e o tipo.

Reply:
pode verificar se existe User  **users?: User;**  , sem isso trás a mensagem de erro [Property 'users' has no initializer and is not definitely assigned in the constructor](https://stackoverflow.com/questions/49699067/property-has-no-initializer-and-is-not-definitely-assigned-in-the-construc)

Reply:
Ele quer dizer que vc precisa inicialiar a propriedade com algum valor. 
Voce pode inicializar ela como `null` e o erro não vai mais ser exibído. Você terá que ajustar o tipo também quando fizer isso. 
```ts
users: User | null = null;
```
Porém eu acho que é um array de users certo ? Você pode tipar ele como `User[]` e inicialar com um array vazio `[]`, e estará perfeito: 
```ts
users: User[] = [];
```

Reply:
Sim, e um um array o users